### DISCUSSION TITLE
External MQTT and MessageBus for single App Service

### DISCUSSION THREAD

**Discussion by zyqhpz** [edgexfoundry/community#239] [Q&A]
Hi, I wanted to know if there's a way to set multiple triggers for a single app service. In my case, I would like to have external MQTT and MessageBus triggers.
 
The use cases are to read real-time data from EdgeX events and send it to external apps through MQTT messages. And get MQTT requests from external apps to process the data through the function pipeline before returning the responses.

As for now, I managed to do it by setting MessageBus as the default trigger and creating a goroutine where it has an MQTT client that listens to external apps. Is there a better way to do this?

**Comment by lenny-goodell** [comment]
Nope, the App SDK allows only one trigger per application service.
